dvc/output/local.py

```diff
@@ -35,14 +35,14 @@ def _parse_path(self, remote, path):
             # FIXME: if we have Windows path containing / or posix one with \
             # then we have #2059 bug and can't really handle that.
             p = self.TREE_CLS.PATH_CLS(path)
-            if not p.is_absolute():
+            if self.stage and not p.is_absolute():
```

Comment:
We should probably remove the dependency on `stage`, this is very much circular: `stage` needs outs/deps, and outs/deps require stages, so, when we construct a stage, there's no way beforehand to create outs/deps without creating a stage first without any deps/outs.

Reply:
@skshetry you mean from the outputs/deps themselves? We still use it in some places, so easier said than done. Logically outs and deps do belong to the stage, but the way we create and fill them is not very friendly, I agree.

Reply:
What I am trying to say, is `Output` should not get access to the `Stage` at all. We could pass `wdir` and `repo` if they need it.